pkg/operator/ceph/disruption/machinelabel/reconcile.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinelabel
+
+import (
+	"context"
+
+	"github.com/coreos/pkg/capnslog"
+	mapiv1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	controllerName                  = "machinelabel-controller"
+	MachineFencingLabelKey          = "fencegroup.rook.io/cluster"
+	MachineFencingNamespaceLabelKey = "fencegroup.rook.io/clusterNamespace"
+	OSDPodLabelValue                = "rook-ceph-osd"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+type ReconcileMachineLabel struct {
+	scheme  *runtime.Scheme
+	client  client.Client
+	options *controllerconfig.Context
+}
+
+type machine struct {
+	isOccupiedByOSD bool
+	RawMachine      mapiv1.Machine
+}
+
+// Reconcile is the implementation of reconcile function for ReconcileMachineLabel
+// which ensures that the machineLabel for the osd pods are in correct state
+func (r *ReconcileMachineLabel) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger.Debugf("reconciling %s", request.NamespacedName)
+
+	// Fetch list of osd pods for the requested ceph cluster
+	pods := &corev1.PodList{}
+	err := r.client.List(context.TODO(), pods, client.InNamespace(request.Namespace),
+		client.MatchingLabels{k8sutil.AppAttr: OSDPodLabelValue, k8sutil.ClusterAttr: request.Name})
```

Comment:
Is there a reason to use a new `OSDPodLabelValue` here instead of reusing the `AppName` parameter in `pkg/operator/ceph/cluster/osd`? That is the purpose of `AppName` yes?

Reply:
good catch